presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java

```diff
@@ -165,7 +165,7 @@ public Optional<String> getJsonRepresentation()
 
     // Serialize this plan fragment with the provided codec, caching the results
     // This should be used when serializing the fragment to send to worker nodes.
-    public synchronized byte[] bytesForTaskSerialization(Codec<PlanFragment> codec)
+    public byte[] bytesForTaskSerialization(Codec<PlanFragment> codec)
```

Comment:
I wonder whether it is better to allow more than one serializer to run concurrently or having a single serializer do the work and let the other consumers wait for it?

Reply:
I dont have strong opinion here. But since this method is synchronized, just accessing the `cachedSerialization` bytes is restricted.. which is a problem. Like once the serialized bytes are ready/cached...access to it should be non-synchronized.

Reply:
That is fair. Ideally multiple task management threads shouldn't be waiting for serialization. What do you think about serializing PlanFragment during planning and then simply read it when scheduling a task?